creator-node/src/routes/tracks.js

```diff
@@ -354,38 +330,50 @@ module.exports = function (app) {
           transaction: t
           }
         )
+
         if (numAffectedRows < trackSegmentCIDs.length) {
           throw new Error('Failed to associate files for every track segment CID.')
         }
-      } else { /** If track updated, ensure files exist with trackuuid. */
-        // Check the transcoded copy if present
-        if (transcodedTrackUUID) {
-          const transcodedFile = await models.File.findOne({
-            where: {
-              fileUUID: transcodedTrackUUID,
-              cnodeUserUUID,
-              trackUUID: track.trackUUID,
-              type: 'copy320'
-            },
-            transaction: t
-          })
-          if (!transcodedFile) {
-            throw new Error('Did not find the corresponding transcoded file for the provided track UUID.')
+
+        // Associate created trackUUID the transcoded copy
+
+        // Using the correct source file, find a transcoded 320kbps copy and associate that
+        const { sourceFile } = trackFiles[0]
+        // Check that each segment file found matches the same source file.
+        // In the case a user uploaded two of the same track twice and failed both times
+        // to associate, this will be exercised. We may want to remove this check here and
+        // let the user recover, but in practice, this may not really happen, so we invariant here
+        // so it's logged.
+        for (let i = 1; i < trackFiles.length; ++i) {
+          if (trackFiles[i].sourceFile !== sourceFile) {
+            throw new Error(
+              `Found non-matching sourcefile ${trackFiles[i].sourceFile} for sourceFile ${sourceFile}`
+            )
           }
         }
 
-        // Check the segment files
-        const trackFiles = await models.File.findAll({
+        const transcodedCopy = await models.File.findOne({
           where: {
-            multihash: trackSegmentCIDs,
+            sourceFile,
             cnodeUserUUID,
-            trackUUID: track.trackUUID,
-            type: 'track'
+            trackUUID: null,
+            type: 'copy320'
+          }
+        })
+
+        const numTranscodeAffectedRows = await models.File.update(
+          { trackUUID: track.trackUUID },
+          { where: {
+            fileUUID: transcodedCopy.fileUUID,
+            cnodeUserUUID,
+            trackUUID: null,
+            type: 'copy320'
           },
           transaction: t
-        })
-        if (trackFiles.length < trackSegmentCIDs.length) {
-          throw new Error('Did not find files for every track segment CID with trackUUID.')
+          })
```

Comment:
nit - vscode doesn't like this spacing. i think it's indented by two spaces

Reply:
not sure why lint didn't catch this. thx!